pyesbulk/__init__.py

```diff
@@ -68,23 +101,58 @@ def put_template(es, name, body):
 
         Failure modes are raised as exceptions.
     """
+    assert name is not None and mapping_name is not None and body is not None
     retry = True
     retry_count = 0
     backoff = 1
     beg, end = time.time(), None
+    try:
+        body_ver = int(body["mappings"][mapping_name]["_meta"]["version"])
+    except KeyError as e:
+        raise Exception(
+            f"Bad template, {name}, given mapping name misssing from template,"
+            f" {mapping_name}: {e!r}"
```

Comment:
Well, OK; but potentially misleading since it could be any of the four key values.

Reply:
Fixed.